src/Microsoft.Diagnostics.Monitoring.WebApi/Stacks/StackData.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.Diagnostics.Monitoring.WebApi.Stacks
+{
+    /// <summary>
+    /// This data model is mostly 1:1 with the information that comes from the EventStacksPipeline.
+    /// Note that most data is either ClassID's or mdToken information.
+    /// </summary>
+    internal sealed class StackResult
+    {
+        public List<Stack> Stacks { get; } = new();
+
+        public NameCache NameCache { get; } = new NameCache();
+    }
+
+    internal sealed class NameCache
+    {
+        public Dictionary<ulong, ClassData> ClassData { get; } = new();
+        public Dictionary<ulong, FunctionData> FunctionData { get; } = new();
+        public Dictionary<ulong, ModuleData> ModuleData { get; } = new();
+        public Dictionary<(ulong moduleId, ulong typeDef), TokenData> TokenData { get; } = new();
+    }
+
+    internal enum ClassFlags : uint
+    {
+        None = 0,
+        Array,
+        Composite,
+        IncompleteData,
+        Error = 0xff
+    }
+
+    internal sealed class ClassData
+    {
+        public ulong[] TypeArgs { get; set; }
+
+        // Because it's possible that we never get the ClassID for a type, we separate Name information from the class.
```

Comment:
I'm confused by this comment...am I missing something, or is this not where it should be?